plugins/metalink/utils.c

```diff
@@ -487,80 +497,86 @@ TDNFMetalinkFree(
 
 uint32_t
 TDNFParseFileTag(
-    TDNF_ML_CTX *ml_ctx,
-    xmlNode *node,
-    const char *filename
+    void *userData
     )
 {
     uint32_t dwError = 0;
-    xmlChar* xmlPropValue = NULL;
     const char *name = NULL;
+    struct MetalinkElementInfo* elementInfo = (struct MetalinkElementInfo*)userData;
 
-    if(!ml_ctx || !node || IsNullOrEmptyString(filename))
+    if(!elementInfo->ml_ctx || IsNullOrEmptyString(elementInfo->filename))
     {
         dwError = ERROR_TDNF_INVALID_PARAMETER;
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
-    xmlPropValue = xmlGetProp(node, ATTR_NAME);
-    if (!xmlPropValue)
+    for (int i=0; elementInfo->attributes[i] ; i+=2)
+    {
+        if(!strcmp(elementInfo->attributes[i], ATTR_NAME))
+        {
+           name = elementInfo->attributes[i+1];
+           break;
+        }
+    }
```

Comment:
Can we make this a function? This can also be used further below.

Reply:
Done